docs/notebooks/Training_visualizations.ipynb

```diff
@@ -0,0 +1,319 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Setup Visdom\n",
+    "\n",
+    "Install it with:\n",
+    "\n",
+    "`pip install visdom`\n",
+    "\n",
+    "Start the server:\n",
+    "\n",
+    "`python -m visdom.server`\n",
+    "\n",
+    "Visdom now can be accessed at http://localhost:8097 in the browser.\n",
+    "\n",
+    "\n",
+    "# LDA Training Visualization\n",
+    "\n",
+    "To monitor the LDA training, a list of Metrics can be passed to the LDA function call for plotting their values live as the training progresses.  \n",
+    "\n",
+    "Let's plot the training stats for an LDA model being trained on Lee corpus. We will use the four evaluation metrics available for topic models in gensim: Coherence, Perplexity, Topic diff and Convergence. (using separate hold_out and test corpus for evaluating the perplexity)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import re\n",
+    "import gensim\n",
+    "from gensim.models import ldamodel\n",
+    "from gensim.corpora.dictionary import Dictionary\n",
+    "\n",
+    "\n",
+    "# Set file names for train and test data\n",
+    "test_data_dir = os.path.join(gensim.__path__[0], 'test', 'test_data')\n",
+    "lee_train_file = os.path.join(test_data_dir, 'lee_background.cor')\n",
+    "lee_test_file = os.path.join(test_data_dir, 'lee.cor')\n",
+    "\n",
+    "def read_corpus(fname):\n",
+    "    texts = []\n",
+    "    with open(fname, encoding=\"ISO-8859-1\") as f:\n",
```

Comment:
Don't work for python2 (because `encoding` isn't supported in python2), replace it to smart_open.